save-demo/src/main/kotlin/com/saveourtool/save/demo/utils/KubernetesUtils.kt

```diff
@@ -0,0 +1,136 @@
+import io.fabric8.kubernetes.client.KubernetesClient
+import io.fabric8.kubernetes.client.KubernetesClientException
+import io.fabric8.kubernetes.client.dsl.ScalableResource
+import org.slf4j.LoggerFactory
+
+private const val DEMO_ORG_NAME = "organizationName"
+private const val DEMO_PROJ_NAME = "projectName"
+private const val DEMO_VERSION = "version"
+private const val REPLICAS_PER_DEMO = 1
+private const val TTL_AFTER_COMPLETED = 3600
+
+private val logger = LoggerFactory.getLogger("KubernetesUtils")
+
+/**
+ * @param demo demo entity
+ * @param kubernetesSettings kubernetes configuration
+ * @throws KubernetesRunnerException on failed job creation
+ */
+@Suppress("NestedBlockDepth")
+fun KubernetesClient.startJob(demo: Demo, kubernetesSettings: KubernetesConfig) {
+    val job = Job().apply {
+        metadata = ObjectMeta().apply {
+            name = jobNameForDemo(demo)
+        }
+        spec = JobSpec().apply {
+            parallelism = REPLICAS_PER_DEMO
+            ttlSecondsAfterFinished = TTL_AFTER_COMPLETED
+            backoffLimit = 0
+            template = PodTemplateSpec().apply {
+                spec = PodSpec().apply {
+                    if (kubernetesSettings.useGvisor) {
+                        nodeSelector = mapOf(
+                            "gvisor" to "enabled"
+                        )
+                        runtimeClassName = "gvisor"
+                    }
+                    metadata = ObjectMeta().apply {
+                        labels = mapOf(
+                            DEMO_ORG_NAME to demo.organizationName,
+                            DEMO_PROJ_NAME to demo.projectName,
+                            DEMO_VERSION to "manual",
+                            // "baseImageName" to baseImageName
+                            "io.kompose.service" to "save-demo-agent",
+                        )
+                    }
+                    // If agent fails, we should handle it manually (update statuses, attempt restart etc.)
+                    restartPolicy = "Never"
+                    containers = listOf(demoAgentContainerSpec(demo.sdk.toSdk().baseImageName(), kubernetesSettings))
+                }
+            }
+        }
+    }
+    logger.debug { "Attempt to create Job from the following spec: $job" }
+    try {
+        resource(job).create()
+        with(demo) {
+            logger.info("Created Job for demo $organizationName/$projectName")
+        }
+    } catch (kex: KubernetesClientException) {
+        with(demo) {
+            throw KubernetesRunnerException("Unable to create a job for demo $organizationName/$projectName", kex)
+        }
+    }
+}
+
+/**
+ * @param demo demo entity
+ * @param namespace namespace that the job should be associated with
+ * @return [ScalableResource] of [Job]
+ */
+fun KubernetesClient.getJob(demo: Demo, namespace: String): ScalableResource<Job> = batch()
+    .v1()
+    .jobs()
+    .inNamespace(namespace)
```

Comment:
do we need to specify it? we already specified it on creating `KubernetesClient`

Reply:
oh sure, will remove it

Reply:
Now do not provide `namespace` but use `KubernetesClient`'s one

Reply:
I see you still provide `namespace`

Reply:
Okok, will remove it